### ISSUE TITLE
Archiver creates invalid archives

### ISSUE THREAD

**Issue by bennymeg** [gregnb/filemanager-webpack-plugin#95]
## Config

```js
new FileManagerPlugin({
  events: {
    onEnd: {
      archive: [
        {
          source: "package.json",
          destination: "package.zip",
        },
      ],
    },
  },
});
```

## Issue

It seems that the archive is being created, and it size reflect the of the files that had being archived, but it cannot be opened.
An file is invalid error is raised when trying to open it.
This happens for both .zip and .tar.gz files.

## Your Environment

| Tech                       | Version    |
| -------------------------- | ---------- |
| filemanager-plugin-webpack | v4.0.0     |
| node                       | v12.21.0   |
| OS                         | windows 10 |

**Comment by sibiraj-s** [comment]
Thanks for reporting this. Can you provide a reproducible code so that I can verify this. I use mac, so far no issues encountered. Could be OS specific. If that is the case, I will need some help fixing this.

**Comment by bennymeg** [comment]
Hi,
I spent couple of hours on that, unfortunately, I could not find find what causing the issue.
It seems to be working fine outside of webpack, I suspect the archive was not properly closed before the webpack process finished.
I have ended up writing a short script to archive the artifacts instead.

**Comment by oliyg** [comment]
I got the same error: 'Unable to expand "app.zip". it is in an unsupported format'
Here is the reproducible [code](https://github.com/oliyg/demozip)
I'm going to make a webpack builder cli application so I choose to use webpack node API.
When my `compiler.close` listener function is triggered I use `process.exit()` to kill the process.
But according to the [archiver doc](https://www.archiverjs.com/docs/archiver/#finalize) although the finalize promise has been settled the destination writing stream has not complete yet.
So maybe I think we should use `close` event to **properly detect stream completion**.

**Comment by sibiraj-s** [comment]
Thanks @oliyg.

Fixed by #107. Published in v6.1.6. Reopen this if the issue still persists.
